src/modules/rover_pos_control/RoverAttitudeControl.cpp

```diff
@@ -0,0 +1,128 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2013-2016 Estimation and Control Library (ECL). All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name ECL nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/**
+ * @file RoverAttitudeControl.hpp
+ * Rover Attitude Controller
+ *
+ * @author Jaeyoung Lim <jaeyoung@auterion.com>
+ *
+ */
+
+#include "RoverAttitudeControl.hpp"
+#include <float.h>
+#include <lib/ecl/geo/geo.h>
+#include <mathlib/mathlib.h>
+
+
+float RoverAttitudeController::control_attitude(const struct ECL_ControlData &ctl_data)
+{
+	//TODO: Move attitude controller from rover pos control
+
+	if (!PX4_ISFINITE(_rate_setpoint)) {
+		PX4_WARN("yaw rate sepoint not finite");
```

Comment:
This needs to be throttled or removed, otherwise the spam might lead to more problems.

Reply:
Is there already an API where I can throttle messages? This is actually coming from https://github.com/PX4/Firmware/blob/master/src/modules/fw_att_control/ecl_pitch_controller.cpp#L56